src/packet_types.c

```diff
@@ -79,3 +80,39 @@ void signal_report_init(SignalReportBlock *b, const int8_t snr, const int8_t rss
     b->contents.request = request;
     b->contents._dead_space = 0;
 }
+/**
+ * Initializes an altitude data block with the provided information.
+ * @param b The altitude data to be initialized
+ * @param measurement_time The mission time at the taking of the measurment
+ * @param pressure The measured pressure in terms of Pascals. This field is a signed 32 bit integer in two's complement
+ * format.
+ * @param temperature The measured temperature in units of 1 millidegree Celsius/LSB. This field is a signed 32 bit
+ * integer in two's complement format.
+ * @param altitude The calculated altitude in units of 1 mm/LSB. This field is a signed 32 bit integer in two’s
+ * complement format.
+ */
+void altidude_data_block_init(AltitudeDataBlock *b, const uint32_t measurement_time, const int32_t pressure,
+                              const uint32_t temperature, const uint32_t altitude) {
+    memcpy(b->bytes, &measurement_time, sizeof(uint32_t));
+    memcpy(b->bytes + 4, &pressure, sizeof(uint32_t));
+    memcpy(b->bytes + 8, &temperature, sizeof(uint32_t));
+    memcpy(b->bytes + 12, &altitude, sizeof(uint32_t));
+}
+/**
+ * Initializes an angular velocity block with the provided information.
+ * @param b The angular velocity block to be initialized.
+ * @param measurement_time The mission time when the measurement was taken.
+ * @param full_scale_range The full scale range of the gyroscope in degrees per second. This value represents the
+ * maximum and minimum angular velocity that can be measured.
+ * @param x_axis The angular velocity measurment for the x axis.
+ * @param y_axis The angular velocity measurment for the y axis.
+ * @param z_axis The angular velocity measurment for the z axis.
+ */
+void angular_velocity_block_init(AngularVelocityBlock *b, const int32_t measurement_time, const int8_t full_scale_range,
+                                 const int16_t x_axis, const int16_t y_axis, const int16_t z_axis) {
+    memcpy(b->bytes, &measurement_time, sizeof(uint32_t));
+    memcpy(b->bytes + 4, &full_scale_range, sizeof(uint16_t)); // Skip 4 bytes where data is already stored
```

Comment:
You should have noticed the compiler warning that says you're trying to write 16 bits from an 8 bit integer. FSR is only 8 bits, which means you should use `sizeof(uint8_t)`. Additionally, you will need to change the `b->bytes + offset` offsets which follow, as they are all off by 1 byte now.